### PR TITLE
Remove duplicate callbacks in ForwardDiff and ReverseDiff adjoints

### PR DESCRIPTION
**Pull request by jClugstor**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API
  
## Additional context
Fixes https://github.com/SciML/SciMLSensitivity.jl/issues/1081

In some of the Adjoint methods, if a problem that was constructed with a callback is passed to it, the callback was being called twice. This because the `kwargs` argument already contains the callback that was passed to the problem constructor. So when the problem is solved, it uses the callback in the problem itself, and the callback passed to the `solve`, effectively doing the same callback operation twice. 

This PR just makes it so that only the callback in the `kwargs` is used.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Needs tests, make the problem from the issue into a test case.

File: test/prob_kwargs.jl [hunk 1]
```diff
@@ -29,3 +29,28 @@ a = ones(3)
 @test Zygote.gradient(f, a)[1][1] ≈ Zygote.gradient(f2, a)[1][1]
 @test Zygote.gradient(f, a)[1][2] == Zygote.gradient(f2, a)[1][2] == 0
 @test Zygote.gradient(f, a)[1][3] == Zygote.gradient(f2, a)[1][3] == 0
+
+# callback in problem construction or in solve call should give same result
+odef(du, u, p, t) = du .= u .* p
```
Comment (ChrisRackauckas) [inline test/prob_kwargs.jl:34]:
```suggestion
# callback in problem construction or in solve call should give same result
# https://github.com/SciML/SciMLSensitivity.jl/issues/1081
odef(du, u, p, t) = du .= u .* p
```

**Comment by ChrisRackauckas** [comment]
Also handle ReverseDiffAdjoint() and TrackerAdjoint as mentioned in the issue.

**Comment by jClugstor** [comment]
It errors out when using ReverseDiffAdjoint or TrackerAdjoint, since the callback tries to mutate the parameter array.

**Comment by ChrisRackauckas** [comment]
Make some `@test_broken` and comment in the original issue. Maybe test another callback that doesn't require mutating?

**Comment by jClugstor** [comment]
Turns out TrackerAdjoint had the same problem.
I made the tests check that the callbacks are called the same number of times, which is the root of the issue.

File: test/prob_kwargs.jl [hunk 2]
```diff
@@ -29,3 +29,37 @@ a = ones(3)
...
+        condition(u, t, integrator) = t == 0.5
+        affect!(integrator) = callback_count2 += 1
+        cb = DiscreteCallback(condition, affect!)
+        prob = ODEProblem{true}(odef, u0p[1:1], (0.0, 1.0), u0p[2:2])
+        sum(solve(prob, Tsit5(), tstops = [0.5], callback = cb, sensealg = adjoint_type))
+    end
+
+    @testset "Callback duplication check" begin
+        for adjoint_type in [
+            ForwardDiffSensitivity(), ReverseDiffAdjoint(), TrackerAdjoint()]
```
Comment (ChrisRackauckas) [inline test/prob_kwargs.jl:57]: and loop through some adjoint methods too, all sensealgs is best

Comment (ChrisRackauckas) [reply test/prob_kwargs.jl:57]: Except ZygoteAdjoint

Comment (jClugstor) [reply test/prob_kwargs.jl]: `ZygoteAdjoint()` and `GaussAdjoint()` error out, besides those I've included all the rest that support callbacks

Comment (ChrisRackauckas) [reply test/prob_kwargs.jl]: For the ones that error, label them as @test_broken. 

Gauss adjoint with callbacks is currently on the critical Todo.

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**